core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala

```diff
@@ -569,7 +569,8 @@ class CassandraReadJournal(system: ExtendedActorSystem, cfg: Config)
       queryPluginConfig.fetchSize,
       None,
       s"currentEventsByPersistenceId-$persistenceId",
-      extractor = Extractors.persistentRepr(eventsByPersistenceIdDeserializer, serialization))
+      extractor = Extractors.persistentRepr(eventsByPersistenceIdDeserializer, serialization),
+      dispatcher = queryPluginConfig.pluginDispatcher)
```

Comment:
Since the user part of the stream run through the public API queries will run on the default dispatcher this means an async boundary is introduced in every app that uses the query side, is that really something we want to do?

Reply:
I'm not sure I understand the previous comment.

Reply:
@johanandren this is not changing that, the dispatchers parameter is delegated down to https://github.com/akka/akka-persistence-cassandra/pull/880/files#diff-279ff4d092baf64b28404c59938d5a8f4053dffe99b8679826391980f66a6f24R647

We must have a separate dispatcher at the inner stage because some operations are blocking.

Reply:
Yes, so if we run that inner `EventsByPersistenceIdStage` and `mapAsync` stage on an internal dispatcher, the user flow running on the default dispatcher will _always_ introduce an async boundary since different dispatchers.

If that's unavoidable I guess that is fine, just that it is understood that is the decision.

Reply:
That's intended, because otherwise we would propagate the blocking out to user responsibility (and it would end up on the default-dispatcher). That doesn't change by this PR, or did I miss something?

See https://github.com/akka/akka-persistence-cassandra/issues/870

Reply:
I see, I guess both solutions has their shortcomings, but async boundary is much better than blocking the default dispatcher indeed. 👍

Reply:
I wonder though if we should make sure all plugin defined stream operator steps run on the plugin dispatcher, so that the only async boundary is between APC stream parts and the user, so that if the user sets a specific dispatcher it won't mean an additional async boundary?

For example: currently in this method with this PR we have `EventByPersistenceStage -> mapAsync` on the plugin internal, then `map -> mapConcat` on default dispatcher,  if the user does `eventsByPersistenceId().yuppeti().yepp().to(Sink.foreach).withAttributes(my-dispatcher)` that will be, perhaps surprisingly, two boundaries.

Reply:
That's a good point. I agree that we should run everything on the plugin dispatcher.
Could you look into that @ignasi35 
I had some problems with the attributes in previous changes. Didn't "bite" at all places. Can be verified with a println inside EventsByTagStage (or the other custom stages):
```
println(Thread.currentThread().getName)
```

Reply:
I think such a thorough review of all streams is beyond the scope of this PR.

Maybe we can merge this, forward port it, and create a separate issue?

Reply:
Adding the plugin dispatcher also for the operators introduced between the internal-internal stages and the internal ops added by the query API should be pretty straight forward, just add that `withAttributes(ActorAttributes.dispatcher(pluginDispatcher))` in the end of each of the query methods you have now passed the dispatcher "inwards" as well.

Verifying is a little bit trickier ofc.

Reply:
@patriknw if it didn't bite and Source.future was used that could be explained by https://github.com/akka/akka/pull/30161 or if `Source.fromMaterialiser/setup` was used https://github.com/akka/akka/pull/30171 

Both are in progress of getting fixed, but would require also setting the attributes on the nested flow to work with current and older Akka versions.

Reply:
> Verifying is a little bit trickier ofc.

Well, that's the problem. I'd rather not modify a handful of lines blindly.

Reply:
That same problem goes for this entire PR though. So it wouldn't be more "blindly" than the current changes?

Reply:
I tried to keep things as small as possible.

Reply:
I only extracted a hardcoded dispatcher name and set the appropriate value in 5 places (CassandraJournal, EventsByTagMigration and CassandreRecovery (x3)).

I got the impression you were suggesting a full review of the whole codebase. It's different orders of magnitude.

Reply:
we should trust that Akka Streams does the right thing so additional automated tests shouldn't be needed here, but since we know there were problems with setting the attributes a manual verification with println is what I was suggesting. I can do that before approving this PR.

Reply:
Actually after reviewing this a bit more in detail I see how tricky it is because of the `futureSource` bug, setting the dispatcher needs to be repeated in lots of places. 

Perhaps it's even preferrable to set the dispatcher where you'd expect to set it (outermost) so that this is fixed once that is fixed in Akka?